paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -230,6 +233,19 @@ def check_expected_slaves(self, slaves, expected_instances):
                 )
                 raise ClusterAutoscalingError(error_message)
 
+    def get_new_timeout_after_terminate(self, slave_to_kill):
+        """Gets a new timeout to set based on the running tasks on a slave about to be killed.
+        The new timeout will be 30s + 30s*(running_tasks), up to 300 seconds.
+        If the slave cannot be found in the state summary, just return the default 300 seconds
+        """
+        state = get_mesos_master().state_summary()
+        timeout = 300
+        for slave in state.get('slaves', []):
```

Comment:
This is an expensive endpoint, can you use an injected version of this instead of fetching it for each slave?

Reply:
The reason I'm fetching it for each slave is that tasks might move around after terminating a slave, especially if this host was previously nearly unused.